stan/math/rev/core/var.hpp

```diff
@@ -34,11 +34,22 @@ template <typename T, typename = void>
 class var_value {};
 
 template <typename T>
-class var_value<T, require_floating_point_t<T>> {
+class var_value<T, require_vt_floating_point<T>> {
+  /**
+   * Checks whether the template type `T` is an assignable type.
+   */
+  template <typename S>
+  using check_plain_type
+      = std::is_same<std::decay_t<S>, std::decay_t<plain_type_t<S>>>;
+  static_assert(
```

Comment:
Nice I want to use these a bunch in `adj_jac_apply` so it's easier to work with.

Reply:
Yeah I'll move this into it's own thing